client/src/components/Workflow/workflows.services.ts

```diff
@@ -105,3 +105,35 @@ export async function getWorkflowInfo(workflowId: string) {
     const { data } = await axios.get(withPrefix(`/api/workflows/${workflowId}`));
     return data;
 }
+
+/**
+ * For dockstore imported IWC workflows, fetch the README file from the GitHub repository.
+ * @param trsId of form `#workflow/github.com/iwc-workflows/<category>/...`
+ * @returns the README file content
+ */
+export async function fetchReadmeForIwcWorkflow(trsId: string) {
+    try {
+        const parts = trsId.split("/");
+        if (parts.length !== 5) {
+            throw new Error("Invalid trsId");
+        }
+
+        // Ensure that the trsID is for a GitHub workflow at iwc-workflows
+        if (parts[0] !== "#workflow" || parts[1] !== "github.com" || parts[2] !== "iwc-workflows") {
+            throw new Error("Workflow is not from iwc-workflows");
+        }
+
+        const category = parts[3];
+
+        // Construct the URL for the README file
+        const readmeUrl = `https://raw.githubusercontent.com/iwc-workflows/${category}/refs/heads/main/README.md`;
```

Comment:
This would be the latest readme, we should have the version if available.

```suggestion
        const readmeUrl = `https://raw.githubusercontent.com/iwc-workflows/${category}/refs/heads/${trs_version_id}/README.md`;
```

Reply:
The version id is stored like: `v0.1.14`

and when i try to fetch 
https://raw.githubusercontent.com/iwc-workflows/allele-based-pathogen-identification/refs/heads/v0.1.14/README.md
for e.g. i get 404

as opposed to `main`

In the metadata, I see the following:
```
{ "trs_server": "dockstore", "trs_tool_id": "#workflow/github.com/iwc-workflows/parallel-accession-download/main", "trs_url": "https://dockstore.org/api/ga4gh/trs/v2/tools/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fparallel-accession-download%2Fmain/versions/v0.1.14", "trs_version_id": "v0.1.14" }
```

Should I instead get that "version" part of the URL from the `trs_tool_id` instead, where we have `.../main` at the end?

Reply:
NVM, I figured it out!
In case there is a `trs_version_id`, that is a github tag on the iwc repo, and so i do this instead:
https://raw.githubusercontent.com/iwc-workflows/allele-based-pathogen-identification/refs/tags/v0.1.2/README.md